config/confighttp/confighttp.go

```diff
@@ -154,8 +160,8 @@ func (hss *HTTPServerSettings) ToServer(handler http.Handler, opts ...ToServerOp
 	for _, o := range opts {
 		o(serverOpts)
 	}
-	if len(hss.CorsOrigins) > 0 {
-		co := cors.Options{AllowedOrigins: hss.CorsOrigins}
+	if len(hss.CorsOrigins) > 0 || len(hss.CorsHeaders) > 0 {
```

Comment:
It seems more intuitive to still enable CORS if there **aren't** any CorsHeaders set and just let it default to the standard set of headers allowed by the [safelist](https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header).

Reply:
@keitwb I didn't want to change flow too much, as I understand in this case we would essentially update CORS by default. I think it makes sense, though unsure if this won't brake something for configurations where it wasn't enabled previously?

Reply:
It would be fully backwards-compatible because the CORS library used in this receiver has a list of headers that are allowed by default  that is based on the CORS standard: https://github.com/rs/cors/blob/master/cors.go#L151.  If somebody doesn't specify CorsHeaders config  it  should default to those headers as  it currently does without the config.

Reply:
Perhaps I am missing something, let me try to share how I understand it:

1. What happened until now was following:

When `cors_allowed_origins` is empty, CORS handler is not enabled
When `cors_allowed_origins` is filled, CORS handler is enabled with other values being defaulted

2. What happens after this change is just a bit more complex:

When both `cors_allowed_origins` and `cors_allowed_headers` are empty, CORS handler is not enabled
When `cors_allowed_origins` has some values, it overwrites [the default](https://github.com/rs/cors/blob/master/cors.go#L123-L128)
When `cors_allowed_headers` has some values, it overwrites [the default](https://github.com/rs/cors/blob/master/cors.go#L151-L154)

So e.g. when someone just provides `cors_allowed_origins` in configuration, then CORS is getting enabled and `AllowedHeaders` is actually set to default value

I think that perhaps we need another attribute to enable it explicitly (`cors_enabled: true`) and then run it with default values or the ones provided, if present. But then that would be a bigger, backward-incompatible change

Reply:
The essence of what it means for CORS to be _enabled_ is that the server respond with a `Access-Control-Allow-Origin: <some non-blank value>` header.   This is the only thing necessary from a server as all the other CORS releated headers have defaults.  This is what  the `cors_allowed_origins` config in this receiver is changing, and so it acts to enable CORS.  Whether you allow non-standard headers or not is a secondary,  non-essential aspect of CORS that needn't be specified for CORS to be enabled.